### PR TITLE
Initial draft of changes

### LINKED ISSUE #3539: The `project delete` command should work with projects that failed to start
When a user wants to delete a project from the CLI they have to pass the project name. The CLI will try to resolve the project's access route to communicate with the orchestrator. To do that translation we just need the `id` field. This information will be retrieved from the `lookup` table from the config file. If it's not found the command will try to fetch the project from the orchestrator.

Currently, the `project delete` command will fail if a project failed to start (i.e. the `Project::is_ready()` never returned `true`). 

We can mitigate this by storing the project data into the config file right before the first loop from `project::util::check_project_readiness()`, at [this line](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/project/util.rs#L135). This way, we will be able to resolve projects id's given their names even if the project failed to start and the user will be able to delete them.

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### PR DESCRIPTION
**Pull request by sanjayts**
- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/project/util.rs [hunk 1]
```diff
@@ -254,9 +257,35 @@ pub mod config {
...
+        } else {
+            None
+        };
+        config.set_project_alias(
+            project.name.to_string(),
+            ProjectLookup {
+                node_route: None,
                 id: project.id.to_string(),
-                identity_id: pid.clone(),
+                identity_id: None,
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/project/util.rs]: You should also pass the node_route and identity_id here based on the `Project`'s values:

```rust
let node_route: Option<MultiAddr> = if project.access_route.is_empty() {
    Some(
        project
            .access_route
            .as_ref()
            .try_into()
            .context("Invalid project node route")?,
    )
} else {
    None
};

...

ProjectLookup {
    node_route,
    identity_id: project.identity_id.clone(),
    ...
}
```

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
I've left a small suggestion, but overall it's looking great @sanjayts. Thank you so much for looking into this :pray:

**Comment by sanjayts** [comment]
@adrianbenavides Awesome, I have made the changes but due to merge conflicts had to create another PR. WDYT?

https://github.com/build-trust/ockam/pull/3636

**Comment by adrianbenavides** [comment]
Sure, no problem, whatever works for you! I'll review it today :+1:

**Pull request closed without merging**